test/rules/strict-boolean-expressions/allow-mix/test.ts.lint

```diff
@@ -0,0 +1,20 @@
+declare function get<T>(): T;
+
+if (get<string | number>()) {}
+    ~~~~~~~~~~~~~~~~~~~~~~ [If statement condition may be falsy if it is 0. Expected a boolean, undefined-union, or string.]
```

Comment:
This error is unclear. How about `This type is not allowed in the 'if' condition because it could be a number. Allowed types are boolean, undefined-union, or string`. You might have to modify it to make the string easier to generate